### ISSUE TITLE
Toggle label is not read at all or read incorrectly in Firefox browser

### ISSUE THREAD

**Issue by ayuspark** [microsoft/fluentui#15109]
### Environment Information

- **Package version(s)**: (fill this out) 7.123.1
- **Browser and OS versions**: (fill this out if relevant) Firefox 80.0.1

### Describe the issue:

When using a toggle, with Windows Narrator or NVDA in the Firefox browser, the label is not read or read incorrectly.

#### Actual behavior:

- Narrator in FF: the label isn't read at all when switching values.
- NVDA in FF: the label was read incorrectly.
![image](https://user-images.githubusercontent.com/23284999/93539667-56d2d880-f906-11ea-8b45-6c2c37263ad4.png)
It reads, "checked" or "not checked" instead.

#### Expected behavior:

- Narrator: it should read the label on each value change.
- NVDA: It should read "Active" or "Inactive"

### If applicable, please provide a codepen repro:

It can be reproed in the Fluent UI's demo page for Toggle

### Priorities and help requested (not applicable if asking question):
Priority 0 for our coming release.
 
Are you willing to submit a PR to fix? (Yes, No)
No
Requested priority: (Blocking, High, Normal, Low)
High
Products/sites affected: (if applicable)

**Comment by paulgildea** [comment]
@ayuspark Thanks for the feedback.

@smhigley Can you help with the desired experience with a screen reader?

@xugao / @JustSlone  I noticed that in the `Toggle` example here: https://developer.microsoft.com/en-us/fluentui#/controls/web/toggle has a specific example called "Enabled and checked (ARIA 1.0 compatible)" where it passes  `role=checkbox`. However, without that example it internally uses `role=switch`. Is it expected that developer need to pass `role=checkbox` to make the component narrate correctly?

**Comment by JustSlone** [comment]
`role="checkbox"` is used for compatibility, `role="switch"` is the right one long term but not supported everywhere yet. So I assume if you need to be ARIA 1.0 compatible you should use `role="checkbox"` per the example.

Regarding narrator not reading the update for the switch in Firefox. It appears that Firefox does not read updates to toggle controls at all (at least in the examples I was able to find)

I tried all these examples and in none of them does Firefox read out the change to the toggle state let alone the label
- https://react-spectrum.adobe.com/react-spectrum/Switch.html
- https://material-ui.com/components/switches/
- https://scottaohara.github.io/aria-switch-control/

@ayuspark I'm not sure if we'll be able to work around the Firefox issue in our library. If you have an example of how it can or should work in Firefox (even just a plain HTML example) that would help a lot, I can't seem to find a working example. 

@smhigley if you have any more info here, please chime in!

**Comment by ayuspark** [comment]
@JustSlone Hi! Thanks for looking into this. I will talk with other team members to see if they have some ideas. I was able to provide solutions for this browser specific issue...

**Comment by smhigley** [comment]
Just wondering: where does the expectation that a screen reader will re-read the accessible name of a control after checking/unchecking it come from? That would not be my expectation -- only some screen readers read accessible name changes, and there's no spec that establishes that as a convention. Nor would I expect a name to re-read after a checked/unchecked state change.

NVDA 2020.1 and FF 80.0.1 work fine for me with the Fluent switch control, and Scott's example at https://scottaohara.github.io/aria-switch-control/ (I didn't try the others). NVDA announces checked/unchecked when I check and uncheck the controls, exactly as expected.

Narrator doesn't really work with Firefox, so I wouldn't worry about what it reads or doesn't read with the switch. Someone using that combo will have much bigger problems :)

**Comment by ayuspark** [comment]
@smhigley Thanks for chiming in.

Scott's example has **on/off** for example, 
![image](https://user-images.githubusercontent.com/23284999/93820787-86d5f080-fc12-11ea-9988-a6ed31f65a5f.png)
and NVDA is reading **checked/not checked**, which is the bug found in our product, that the reading didn't match the label.

What's your thoughts here?

**Comment by smhigley** [comment]
In that example, the label is "Use large fonts", which is reading OK. The strings "on" and "off" aren't labels, but text alternatives for the state (checked vs. unchecked). I would not expect NVDA (or any screen reader) to read them.

**Comment by ayuspark** [comment]
> In that example, the label is "Use large fonts", which is reading OK. The strings "on" and "off" aren't labels, but text alternatives for the state (checked vs. unchecked). I would not expect NVDA (or any screen reader) to read them.

Okay, let's go back to fluent UI then :)
for the following switch screenshots:
![image](https://user-images.githubusercontent.com/23284999/93821976-51ca9d80-fc14-11ea-8082-6f48880ea32a.png)
![image](https://user-images.githubusercontent.com/23284999/93821997-5c853280-fc14-11ea-9236-2aef02948f56.png)

It appears to be a label tag
![image](https://user-images.githubusercontent.com/23284999/93822143-9b1aed00-fc14-11ea-8eb9-0390fbe7f1f1.png)

Our product was pointed out that NVDA wasn't announcing the correct label in FF. (In Chrome and Edge, it's correct).

It's not reading **"Active"/"Inactive"**, but instead, "Checked"/"Not checked".

**Comment by smhigley** [comment]
Even in that case, the active/inactive strings are still just alt text for the state. The accessible name of the switch control should be whatever the actual label text is for that control.

A good rule of thumb is if you can't replace the inline state text with "on"/"off" or remove it entirely without sacrificing meaning, then that text is probably in the wrong place.

**Comment by scottaohara** [comment]
> It's not reading "Active"/"Inactive", but instead, "Checked"/"Not checked".

this is not a bug, but rather how switches/checkboxes should behave.  state announcement should come from `aria-checked` on these roles.  The visible text string of that state really shouldn't be announced at all, as it just adds redundant state noise.

Testing the fluent UI toggle switches, I see that the visible state text has been added to the accessible name, so the first example can read as "enabled and checked on, switch, on".  Is there a reason this was added, aside from trying to mitigate misunderstandings of what the label is, such as this issue?  cc @smhigley 


Note: how far back into legacy browser versions does Fluent support?  At least back to 2018 an element with `role=switch` will either announce with a "switch" role, or it will be exposed with a "checkbox" role (which, is just how some AT have decided to announce this role).
